### PR TITLE
(#1475) add show my idea checkbox

### LINKED ISSUE #1499: fix: double user ideas when toggling checkbox
When toggling the checkbox on and off, user ideas will stay at the very top and displayed twice on the list. Existing ideas that supposed to be there are not shown although it exists on the array.

### PR DESCRIPTION
**Pull request by rz-p**
Will close #1499 and #1475 

**Todo:**
- [x] Add checkbox to archive
- [ ] ~~When disabling checkbox, the top ideas are still the userIdeas & still shown at the bottom of the list (double occurence)~~

### CODE REVIEW

**Comment by rz-p** [comment]
**Existing Bug**
When toggling the checkbox on and off, user ideas will stay at the very top and displayed twice on the list. Existing ideas that supposed to be there are not shown although it exists on the array.

Will be addressed in ticket ....

File: hms-app/src/pages/AllIdeas.tsx [hunk 1]
```diff
@@ -62,22 +72,21 @@ function AllIdeas() {
...
+  // Disabled search bar for now
+  // const searchIdea = relevantIdeaList.filter((item) => {
+  //   return item.title?.toLowerCase().includes(searchTerm.toLowerCase())
+  // })
+
+  const userIdea = relevantIdeaList.filter((item) => {
+    const userId = user?.id
+    return (
+      item.owner?.id === userId &&
+      item.title?.toLowerCase().includes(searchTerm.toLowerCase())
```
Comment (LetsPoke) [inline hms-app/src/pages/AllIdeas.tsx:88]: why do we use the searchTerm if we dont use the search?

File: hms-app/src/pages/AllIdeas.tsx [hunk 2]
```diff
@@ -325,8 +350,29 @@ function AllIdeas() {
...
+                <Stack align='flex-start' justify='flex-start' spacing='sm'>
+                  {showUserIdeas ? (
+                    <Title order={2} mt={50}>
+                      Your submission: {userIdeaList.length}
+                    </Title>
+                  ) : (
+                    <Title order={2} mt={50}>
+                      Ideas submitted: {relevantIdeaList.length}
+                    </Title>
+                  )}
```
Comment (verzano) [inline hms-app/src/pages/AllIdeas.tsx:362]: you could put the ternary inside the `<Title order...` tags to eliminate a little duplication

**Comment by rz-p** [comment]
Archive page won't show the ideas from past hackathons

**Comment by LetsPoke** [comment]
after update from main it looks soooo far so good 
except: 
- archiv checkbox doesnt work
- hackathons move idea doesnt update the idea list after idea got moved

**Comment by rz-p** [comment]
Branch deprecated

**Pull request closed without merging**